src/smpi/smpi_shared.cpp

```diff
@@ -218,23 +223,63 @@ static void *smpi_shared_malloc_local(size_t size, const char *file, int line)
  */
 void* smpi_shared_malloc_partial(size_t size, size_t* shared_block_offsets, int nb_shared_blocks)
 {
-  void *mem;
-  xbt_assert(smpi_shared_malloc_blocksize % PAGE_SIZE == 0, "The block size of shared malloc should be a multiple of the page size.");
+  char *huge_page_mount_point = xbt_cfg_get_string("smpi/shared-malloc-hugepage");
+  const bool use_huge_page = huge_page_mount_point[0] != '\0';
+#ifndef MAP_HUGETLB
```

Comment:
Please don't use #ifndef, because it makes it hard to detect whether you made a typo in the name (or removed a needed header file). Instead, use `#if !MAP_HUGETLB` or similar. With this, the `-Wundef` that we are using can detect the previous errors.